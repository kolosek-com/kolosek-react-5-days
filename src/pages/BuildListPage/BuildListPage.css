.circle-ci-build-list {
  max-width: 1000px;
  margin: 50px auto;
  padding: 50px;
  border: 1px solid #000000;
}
