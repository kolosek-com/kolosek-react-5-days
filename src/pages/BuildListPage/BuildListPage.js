import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Redirect } from 'react-router-dom';
import ProjectSelect from '../../components/ProjectSelect/ProjectSelect';
import BranchSelect from '../../components/BranchSelect/BranchSelect';
import CircleCiBuild from '../../components/CircleCiBuild/CircleCiBuild';

import './BuildListPage.css';

import {
  getBranchBuildList,
  getProjectBuildList,
  getBuildList,
  selectBuild,
} from '../../reducers/BuildReducer/actions';

class BuildListPage extends Component {
  componentDidMount() {
    this.fetchBuilds(this.props.project, this.props.branch);
  }

  fetchBuilds = (project, branch) => {
    if (project && branch) {
      this.props.getBranchBuildList(this.props.apiKey, project, branch);
    } else if (project) {
      this.props.getProjectBuildList(this.props.apiKey, project);
    } else {
      this.props.getBuildList(this.props.apiKey);
    }
  }

  selectBuild = (build) => () => {
    this.props.selectBuild(build);
  }

  selectProject = (project) => {
    this.fetchBuilds(project, null);
  }

  selectBranch = (branch) => {
    this.fetchBuilds(this.props.project, branch.length > 0 ? branch : null);
  }

  render() {
    const { apiKey } = this.props;
    if (!apiKey) {
      return (
        <Redirect
          to='/login'
        />
      )
    }
    return (
      <div className="circle-ci-build-list">
        <div className="row">
          <div className="col-12 text-center">
            <h1>Circle CI</h1>
          </div>
        </div>
        <div className="row">
          <div className="col-12">
            <h4>Latest Builds</h4>
          </div>
        </div>
        <div className="row">
          <div className="col-3">
            <ProjectSelect onSelect={this.selectProject} />
          </div>
          <div className="col-9">
            <BranchSelect onSelect={this.selectBranch} />
          </div>
        </div>

        {
          this.props.builds.map((build, index) => {
            return (<CircleCiBuild build={build} onClick={this.selectBuild(build)} />);
          })
        }
      </div>
    );
  }
}

function stateToProps(state) {
  return {
    apiKey: state.apiKey.currentKey,
    builds: state.builds.buildList,
    project: state.projects.selectedProject,
    branch: state.projects.selectedbranch,
  };
}

function dispatchToProps(dispatch) {
  return bindActionCreators({
    getBranchBuildList,
    getProjectBuildList,
    getBuildList,
    selectBuild,
  }, dispatch);
}

const BuildListPage_Connected = connect(stateToProps, dispatchToProps)(BuildListPage);

export default BuildListPage_Connected;
