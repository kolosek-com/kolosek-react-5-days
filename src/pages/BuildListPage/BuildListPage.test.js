import React from 'react';
import renderer from 'react-test-renderer';
import { BuildListPage } from './BuildListPage'
import { BrowserRouter as Router }    from 'react-router-dom';

test('BuildListPage with selectedBuild', () => {
  const builds = [{
    branches: {
      development: {
        recent_builds: [{
          added_at: "2018-08-29T09:38:01.019Z",
          build_num: 3,
          is_2_0_job: true,
          is_workflow_job: false,
          outcome: "failed",
          pushed_at: "2018-08-29T09:37:41.548Z",
          status: "failed",
          vcs_revision: "29dc0e38446d5e4a7030caa7e61424b4aee7e175"
        }]
      }
    }
  }]

  const isAuthenticated = jest.fn()
  isAuthenticated.mockReturnValue(true);

  const getCircleciBuilds = jest.fn()
  getCircleciBuilds.mockReturnValue(builds)

  const component = renderer.create(
    <Router>
      <BuildListPage builds={builds} isAuthenticated={isAuthenticated} getCircleciBuilds={getCircleciBuilds} />
    </Router>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
