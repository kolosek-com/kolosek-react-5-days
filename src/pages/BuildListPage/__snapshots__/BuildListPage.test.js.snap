// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildListPage with selectedBuild 1`] = `
<div>
  <h1>
    CircleCI
  </h1>
  <h2>
    Latest Builds
  </h2>
  <div>
    <a
      href="/build/29dc0e38446d5e4a7030caa7e61424b4aee7e175"
      onClick={[Function]}
    >
      <h3>
        development
         / 
         #
        3
         
        2018-08-29T09:38:01.019Z
         time ago
      </h3>
      <p>
        Status: 
        failed
      </p>
    </a>
  </div>
</div>
`;
